app/src/main/java/com/teamltt/carcare/activity/LoginActivity.java

```diff
@@ -133,13 +133,42 @@ private void handleSignInResult(GoogleSignInResult result) {
         Log.d(TAG, "handleSignInResult:" + result.isSuccess());
         if (result.isSuccess()) {
             // Signed in successfully, show authenticated UI.
-            googleSignInAccount = result.getSignInAccount();
-            tvStatus.setText(getString(R.string.signed_in_fmt, googleSignInAccount.getDisplayName()));
-            updateUI(true);
+            GoogleSignInAccount googleSignInAccount = result.getSignInAccount();
+            if (googleSignInAccount != null) {
+                tvStatus.setText(getString(R.string.signed_in_fmt, googleSignInAccount.getDisplayName()));
+
+                String firstName = googleSignInAccount.getGivenName();
+                String lastName = googleSignInAccount.getFamilyName();
+                String googleId = googleSignInAccount.getId();
+
+                if (!addUserIfNew(googleId, firstName, lastName)) {
+                    Log.e(TAG, "could not create new user");
+                }
+
+                // Go to Home Screen
+                Intent intent = new Intent(this, HomeActivity.class);
```

Comment:
I would consider not doing this if the above "could not create new user" condition triggers, but I understand we may not want that.